data_structures/dict_text_analysis.py

```diff
@@ -0,0 +1,24 @@
+# dictionaries for text analysis in a file.
+
+import os
+
+
+def text_analysis(file_name):
+
+    file1 = open(file_name, "r")
+    d = dict()
```

Comment:
Use a defaultdict(int) so that you don't have to do the if check on line 13-16.

Reply:
dict() replaced with defaultdict(init)